tex/latex/pgf-umlcd/pgf-umlcd.sty

```diff
@@ -111,6 +111,19 @@ minimum height=1cm, node distance=2cm]
 \end{classAndInterfaceCommon}
 }
 
+\newenvironment{enumeration}[3][]%
+{
+\begin{classAndInterfaceCommon}{#1}{#2}{#3}
+}%
+{\calcuateNumberOfParts{}
+\node[this umlcd style, anchor=north] (\umlcdClassName) at (\umlcdClassPos)
+    {$<<$enumeration$>>$ \\ \textbf{\umlcdClassName}
+\insertAttributesAndOperations{}
+};
+
+\end{classAndInterfaceCommon}
+}
+
```

Comment:
There are a lot of potential spurious spaces and the indentation (or rather the lack thereof) is suboptimal.

```suggestion
\newenvironment{enumeration}[3][]%
{%
  \begin{classAndInterfaceCommon}{#1}{#2}{#3}%
}%
{%
    \calcuateNumberOfParts{}%
    \node[this umlcd style, anchor=north] (\umlcdClassName) at (\umlcdClassPos)
      {$<<$enumeration$>>$\\ \textbf{\umlcdClassName}\insertAttributesAndOperations{}};%
  \end{classAndInterfaceCommon}%
}

```